feat: add types for afr

pleshevskiy opened the pull request:
Me again with my types, but this time for afr :) This time I tried to take your style into account :)

pleshevskiy committed:
feat: add types for afr

pleshevskiy committed:
fix: res type for overrided method

pleshevskiy committed:
chore: expose all types

mitranim commented:
Nice! Glad you find this useful. Actually, since Afr is Deno-only, it could just be converted to TS, without the need for additional definitions. The only reason it isn't, is because Node support was dropped only recently. I could probably find the time... not sure how long this takes...

pleshevskiy commented:
I can change to TS in this PR

mitranim commented:
Oh? That could be quite nice 🙂

Since Afr is usable from Deno JS code, which doesn't get typechecked, we should keep all the type assertions, in addition to the static checks. Basically I think no redesign is needed, just extra annotations 🙂

pleshevskiy committed:
refac: add ts version

mitranim committed:
TS: revision and corrections

`contentTypes`: restore to null-prototype to avoid collisions with inherited
`Object` properties and methods; restore to mutable.

`watch`: more flexible target signature (match Deno).

Minor bugfix in `contentType`: always convert to pathname, making sure to ignore
search/hash.

`Dir`: remove unnecessary encapsulation of properties.

`Broad`: remove unnecessary encapsulation of properties.

`Broad.prototype.send`: fix a bug introduced by the separation of `write` and
`writeJson` in `ReadWriter`.

`main`: more sensible port detection.

`ReadWriter`: less hacky constructor implementation.

`BroadClient`: remove unnecessary encapsulation of properties.

`readableStreamFromReader`: make `Closer` optional to more closely match the
Deno stdlib implementation of this function.

Type-checking utils: revise and simplify type signatures.

Dedup some repeating types.

Rename some types and methods for clarity and brevity.

Drop now-unused `afr.mjs`.

Drop some unnecessary punctuation.

pleshevskiy committed:
fix: deno any types

mitranim committed:
type corrections for Deno linter

mitranim committed:
revert incorrect changes in `procure`

mitranim committed:
remove lefover booleans from `dirResolveSiteFile`
